cuda_core/cuda/core/experimental/_memory/_buffer.pyx

```diff
@@ -55,13 +56,14 @@ cdef class Buffer:
     @classmethod
     def _init(
         cls, ptr: DevicePointerT, size_t size, mr: MemoryResource | None = None,
-        stream: Stream | None = None
+        stream: Stream | None = None, ipc_descriptor: IPCBufferDescriptor | None = None
     ):
         cdef Buffer self = Buffer.__new__(cls)
         self._ptr = <uintptr_t>(int(ptr))
         self._ptr_obj = ptr
         self._size = size
         self._memory_resource = mr
+        self._ipc_data = IPCDataForBuffer(ipc_descriptor, True) if ipc_descriptor is not None else None
```

Comment:
Optional: This would be more readable with `is_mapped=True`.

I'd do that systematically for all `IPCDataForBuffer` and `IPCDataForMR` calls (I think there are only four calls).